static/app/components/draggableTabs/draggableTabList.tsx

```diff
@@ -24,6 +38,202 @@ interface BaseDraggableTabListProps extends DraggableTabListProps {
+      trigger={triggerProps => (
+        <OverflowMenuTrigger
+          {...triggerProps}
+          size="sm"
+          borderless
+          showChevron={false}
+          icon={<IconEllipsis />}
+          aria-label={t('More tabs')}
+        />
+      )}
+    />
+  );
+}
+
+function Tabs({
+  orientation,
+  ariaProps,
+  state,
+  className,
+  onReorder,
+  tabVariant,
+  setTabRefs,
+  tabs,
+  overflowingTabs,
+}: {
+  ariaProps: AriaTabListOptions<DraggableTabListItemProps>;
+  onReorder: (newOrder: Node<DraggableTabListItemProps>[]) => void;
+  orientation: 'horizontal' | 'vertical';
+  overflowingTabs: Node<DraggableTabListItemProps>[];
+  setTabRefs: Dispatch<SetStateAction<Array<HTMLDivElement | null>>>;
+  state: TabListState<DraggableTabListItemProps>;
+  tabs: Node<DraggableTabListItemProps>[];
+  className?: string;
+  disabled?: boolean;
+  onChange?: (key: string | number) => void;
+  tabVariant?: BaseTabProps['variant'];
+  value?: string | number;
+}) {
+  const tabListRef = useRef<HTMLUListElement>(null);
+  const {tabListProps} = useTabList({orientation, ...ariaProps}, state, tabListRef);
+
+  const values = useMemo(() => [...state.collection], [state.collection]);
+
+  const isDragging = useIsDragging(tabListRef);
+
+  // Only apply this while dragging, because it causes tabs to stay within the container
+  // which we do not want (we hide tabs once they overflow
+  const dragConstraints = isDragging ? tabListRef : undefined;
+
+  return (
+    <TabListWrap {...tabListProps} className={className} ref={tabListRef}>
+      <ReorderGroup axis="x" values={values} onReorder={onReorder} as="div">
+        {tabs.map((item, i) => (
+          <Fragment key={item.key}>
+            <TabItemWrap
+              isSelected={state.selectedKey === item.key}
+              ref={el =>
+                setTabRefs(old => {
+                  if (!el || old.includes(el)) {
+                    return old;
+                  }
+
+                  const newRefs = [...old];
+                  newRefs[i] = el;
+                  return newRefs;
+                })
+              }
+              value={item}
+              as="div"
+              data-key={item.key}
+              dragConstraints={dragConstraints} // dragConstraints are the bounds that the tab can be dragged within
+              dragElastic={0} // Prevents the tab from being dragged outside of the dragConstraints (w/o this you can drag it outside but it'll spring back)
+              dragTransition={{bounceStiffness: 400, bounceDamping: 40}} // Recovers spring behavior thats lost when using dragElastic=0
+              transition={{delay: -0.1}} // Skips the first few frames of the animation that make the tab appear to shrink before growing
+              layout
+            >
+              <div
+                key={item.key}
+                style={
+                  overflowingTabs.some(tab => tab.key === item.key) ? {opacity: 0} : {}
+                }
```

Comment:
Just curious - why can't we apply this style directly to the child Tab component?

Reply:
Ah good catch, this style wasn't necessary. Left from when I was testing something earlier.